packages/overlay/src/overlay-stack.ts

```diff
@@ -585,19 +593,27 @@ export class OverlayStack {
         if (this.preventMouseRootClose || event.defaultPrevented) {
             return;
         }
-        this.closeTopOverlay();
-        const overlaysToClose = [];
-        let root: HTMLElement | undefined = this.topOverlay?.root;
-        let overlay = parentOverlayOf(root);
-        while (root && overlay) {
-            overlaysToClose.push(overlay);
-            overlay = parentOverlayOf(root);
-            root = overlay?.root;
+        if (this.topOverlay?.interaction === 'hover') {
+            // Don't close the top overlay on click if it comes from a hover, it will be closed when the mouse is moved out
+            // But do close everything that came from a click since we just clicked elsewhere!
+            this.closeAllClickOverlays();
```

Comment:
I wonder if this should just be the default behavior for a click event, regardless of whether the top overlay is a hover overlay...

Reply:
In that it's hard to have a hover that you're not hovered on, seems like it might not be a bad idea. Though it would really be `closeAllNonHoverOverlays()`, I think.

Reply:
Ok, I made that change... But I wonder, will this close dialogs randomly when you click on things? That seems wrong... And I imagine it could have happened before if the top overlay was within a Dialog... There might be a lot more logic necessary here.